From: Ross Vandegrift <ross@kallisti.us>
Date: Sat, 29 Mar 2014 20:12:18 +0000
Subject: Fix rawapi tests and enable them

When testing option getting, we need to check the option capabilities
for their active status.  Inactive options cannot be read.

Signed-off-by: Ross Vandegrift <ross@kallisti.us>
---
 run_tests.py          | 8 ++++----
 tests/tests_rawapi.py | 2 ++
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/run_tests.py b/run_tests.py
index bf427e7..04a525c 100755
--- a/run_tests.py
+++ b/run_tests.py
@@ -10,10 +10,10 @@ from src import abstract
 from src import abstract_th
 
 if __name__ == '__main__':
-    #print("---")
-    #print("=== RawAPI: ===")
-    #unittest.TextTestRunner(verbosity=3).run(
-    #    tests_rawapi.get_all_tests())
+    print("---")
+    print("=== RawAPI: ===")
+    unittest.TextTestRunner(verbosity=3).run(
+       tests_rawapi.get_all_tests())
     print("---")
     print("=== Abstract: ===")
     unittest.TextTestRunner(verbosity=3).run(
diff --git a/tests/tests_rawapi.py b/tests/tests_rawapi.py
index 4fef1e2..3723548 100644
--- a/tests/tests_rawapi.py
+++ b/tests/tests_rawapi.py
@@ -107,6 +107,8 @@ class TestSaneControlOption(unittest.TestCase):
             desc = rawapi.sane_get_option_descriptor(self.dev_handle, opt_idx)
             if not rawapi.SaneValueType(desc.type).can_getset_opt():
                 continue
+            if desc.cap|rawapi.SaneCapabilities.INACTIVE == desc.cap:
+                continue
             val = rawapi.sane_get_option_value(self.dev_handle, opt_idx)
             self.assertNotEqual(val, None)
 
